### ISSUE TITLE
Org mode auto-markup plugin

### ISSUE THREAD

**Issue by agent725** [micro-editor/plugin-channel#29]
Hi,

Micro editor is a great tool. I am trying to make a plugin to integrate Org Mode into it using pandoc. What I cannot figure out is two things:

    How to save the text buffer to a temporary file.
    How to load the text of the temporary file into the current buffer, without opening a new buffer.

Can you help me out? I can give you a nice plugin in return. ;-)

Cheers,
Agent725

**Comment by JohnGraber** [comment]
Adding org mode to micro would be a massive contribution, in my opinion.

I hope someone can answer these questions for you!

**Comment by vv52** [comment]
Is this project still ongoing?

I've been looking into porting org mode to micro or otherwise finding a way to wrap emacs to provide org functionality in micro.

If any of the work has been done and abandoned, I would be happy to take over the project.

**Comment by agent725** [comment]
No, I did not get the information needed to continue. However, please feel free to pick it up, and I'd help where I can.

**Comment by taconi** [comment]
Hello, if there is anyone who still wanted to implement this functionality, could you please comment?

I can try to help resolve the issues being faced.

I think of some plugins that implement features such as saving texts in temporary files and writing this text to the current buffer and that could be taken to study:

- [scratch](https://github.com/samdmarshall/micro-scratch-plugin) creates a temporary file
- [manager](https://codeberg.org/micro-plugins/manager) saves text to a file that represents the state of a given plugin function and saves recently opened files/projects to list
- [snippets](https://github.com/micro-editor/updated-plugins/tree/master/micro-snippets-plugin) writes to current file based on text saved in another file
- [lsp](https://github.com/AndCake/micro-plugin-lsp) and [mlsp](https://github.com/Andriamanitra/mlsp) writes to the current file to autocomplete

There is also the [microwiki](https://github.com/obedm503/microwiki) plugin which I haven't used yet, it seems to be really cool and can perhaps be used instead of an orgmode.

**Comment by Naheel-Azawy** [comment]
Hello, I still do want this :)

**Comment by taconi** [comment]
@Naheel-Azawy Do you need help with anything to implement this functionality?

**Comment by Naheel-Azawy** [comment]
@taconi Well, I meant I still want to have an org mode plugin for micro. But unfortunately I don't think I can allocate time to do this myself any time soon.
I just meant I'll be very grateful if someone implement this.

**Comment by taconi** [comment]
@Naheel-Azawy Don't the [microwiki](https://github.com/obedm503/microwiki) or [microzim](https://github.com/ihavenomouth/microzim) plugins cover this?

**Comment by Naheel-Azawy** [comment]
No. vimwiki and zimwiki have a different syntax compared to orgmode.
But I think they could be a good start if someone starts from these and then create an orgmode specific plugin.

**Comment by taconi** [comment]
[orgmode](https://orgmode.org/) is an [emacs mode](https://www.gnu.org/software/emacs/manual/html_node/emacs/Modes.html), [emacs](https://www.gnu.org/software/emacs/) is not a terminal editor like `micro` so it would not be possible to implement all the functionality that exists there.
Maybe you could use [nvim-orgmode](https://github.com/nvim-orgmode/orgmode) as a base.

**Comment by ScottFreeCode** [comment]
I don't think Micro needs most of Emacs's Org Mode features. If I want full Org Mode and all its extra functionality and all its customizable extensions, I'll use Emacs. For Micro I'd settle for syntax highlighting and whatever else Micro's Markdown support has.

That way I would generally not stuck using Markdown or switching over from Micro to Emacs until I actually need Emac's Org programmability or the extra features built on it.

I guess individual Org Mode features like, I dunno, tangle/babel, or agenda views, or scheduling/timekeeping, might be nice to implement as plugins of their own (one per feature). That potentially gets complicated depending on whether supporting them requires reading or acting on any Emacs Lisp…

Just my two cents' worth.
